### PR TITLE
Update on Swiss Glaciers

### PR DESCRIPTION
**Pull request by marvande**
MassBalanceMachine:
- Correction of computation of glacier grids in `Dataset.py` and  `create_glacier_grid.py`
- Add Hugonnet dhdt to [get_topo_data.py](https://github.com/ODINN-SciML/MassBalanceMachine/compare/main...marvande:MassBalanceMachine:main?expand=1#diff-8b1be3401101f0e02275fbd50066d7e75e9df233f5d4b4cf717927ce70f7911f)
- Add possibility to calculate evaluation score to a specific seasonal period (e.g., over winter MB) in [CustomXGBoostRegressor.py](https://github.com/ODINN-SciML/MassBalanceMachine/compare/main...marvande:MassBalanceMachine:main?expand=1#diff-50ab361bd027e6a9632396b95b7b6a6329f27b7aecec14255a751a7646157ce1)

CH analysis: 
- Computation of MB over whole grid of a glacier
- Blocking on glaciers for CH

### CODE REVIEW

File: config.py
Comment (khsjursen) [inline config.py:1]: Should this be set to default?

File: massbalancemachine/data_processing/Dataset.py [hunk 1]
```diff
@@ -157,9 +154,13 @@ def create_glacier_grid(self,
         ds, glacier_indices, gdir = get_glacier_mask(self.data,
                                                      custom_working_dir)
         years_stake = self.data['YEAR'].unique()
-        years = range(years_stake.min(), years_stake.max() + 1)
+        years = range(1951, 2023)
```
Comment (khsjursen) [inline massbalancemachine/data_processing/Dataset.py:157]: Why the fixed time range? Maybe comment on that if it is a quick fix for something that is not working?

File: massbalancemachine/data_processing/get_climate_data.py
Comment (khsjursen) [inline massbalancemachine/data_processing/get_climate_data.py:1]: Why did you change this?

File: massbalancemachine/data_processing/Dataset.py (hunk 1 above)
Comment (marvande) [reply massbalancemachine/data_processing/Dataset.py:157]: It's because we want the grid from the beginning of ERA5 to the end. The way it was done before it took the min and max of measurement years, but I actually also want to create glacier grids beyond that.
